### PR TITLE
Increased timeout and separated out nats invalid sink tests

### LINKED ISSUE #12813: NATS subscription controller tests are flaky
**Description**

Our NATS subscription controller tests, which used to be pretty stable, seems to be flaky after #12208. This is based on the post-commit jobs that run our unit tests.

https://status.build.kyma-project.io/view/gs/kyma-prow-logs/logs/post-kyma-components-eventing-controller/1466071949774950400
https://status.build.kyma-project.io/view/gs/kyma-prow-logs/logs/post-kyma-components-eventing-controller/1467851684695773184

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Increased timeout and separated out nats invalid sink tests

**Related issue(s)**

`Fixes #12813`

### CODE REVIEW

File: components/eventing-controller/controllers/subscription/nats/reconciler_test.go [hunk 1]
```diff
@@ -117,160 +117,79 @@ func testCreateDeleteSubscription(id int, eventTypePrefix, natsSubjectToPublish,
 }
 
 func testCreateSubscriptionWithInvalidSink(id int, eventTypePrefix, natsSubjectToPublish, eventTypeToSubscribe string) bool {
```
Comment (FriedrichWilken) [inline components/eventing-controller/controllers/subscription/nats/reconciler_test.go:119]: You are no longer checking this in your test:
```
Expect(backendSubscription.Subject).Should(Equal(natsSubjectToPublish))
```
if this is ententional you could change the functions signature  to
```suggestion
func testCreateSubscriptionWithInvalidSink(id int, eventTypePrefix, _, eventTypeToSubscribe string) bool {
```
and still fulfill the `testCase` requirements

**Review by FriedrichWilken — COMMENTED** [review]
You have removed the check
```
Expect(backendSubscription.Subject).Should(Equal(natsSubjectToPublish))
```
I am not sure if this is really no longer needed. Can you explain?

Overall, this test seems to be reliable now and this is a great improvement for code quality and readability.

**Comment by mfaizanse** [comment]
> You have removed the check
> 
> ```
> Expect(backendSubscription.Subject).Should(Equal(natsSubjectToPublish))
> ```
> 
> I am not sure if this is really no longer needed. Can you explain?
> 
> Overall, this test seems to be reliable now and this is a great improvement for code quality and readability.

I removed the whole section below `By("Updating the subscription configuration in the spec with invalid sink URL with valid subscriber service name")` and this line was also part of this section.
The reason for removal was that the correct sink case is already tested/covered in the other test `Should create/delete NATS Subscription`.

**Comment by FriedrichWilken** [comment]
> > You have removed the check
> > ```
> > Expect(backendSubscription.Subject).Should(Equal(natsSubjectToPublish))
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I am not sure if this is really no longer needed. Can you explain?
> > Overall, this test seems to be reliable now and this is a great improvement for code quality and readability.
> 
> I removed the whole section below `By("Updating the subscription configuration in the spec with invalid sink URL with valid subscriber service name")` and this line was also part of this section. The reason for removal was that the correct sink case is already tested/covered in the other test `Should create/delete NATS Subscription`.

Thanks. I missed that the check was part of the removed 'By'.

**Review by FriedrichWilken — APPROVED** [review]

**Review by k15r — APPROVED** [review]

File: components/eventing-controller/controllers/subscription/nats/reconciler_test.go [hunk 2]
```diff
@@ -116,161 +116,80 @@ func testCreateDeleteSubscription(id int, eventTypePrefix, natsSubjectToPublish,
...
-			Expect(k8sClient.Update(ctx, changedSub)).Should(BeNil())
-
-			getSubscription(ctx, givenSubscription).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveCondition(eventingv1alpha1.MakeCondition(
-					eventingv1alpha1.ConditionSubscriptionActive,
-					eventingv1alpha1.ConditionReasonNATSSubscriptionActive,
-					v1.ConditionTrue, "")),
-			))
+func testCreateSubscriptionWithInvalidSink(id int, eventTypePrefix, _, eventTypeToSubscribe string) bool {
```
Comment (radufa) [inline components/eventing-controller/controllers/subscription/nats/reconciler_test.go:119]: Perhaps we should have a "positive" test too, something like:
```
 getSubscriptionWithValidSink(...)
 ```
 which should avoid issues like this: https://github.com/kyma-project/kyma/issues/12810
 If you think that's another issue, please create a new one and link it

Comment (mfaizanse) [reply components/eventing-controller/controllers/subscription/nats/reconciler_test.go:119]: @radufa positive test is already covered in this [test](https://github.com/kyma-project/kyma/blob/d91cde56f0bda1504e4a2e9e8869f822c5e49823/components/eventing-controller/controllers/subscription/nats/reconciler_test.go#L93)

**Pull request merged**